### PR TITLE
Library refacto

### PR DESCRIPTION
**Pull request by qparis**

### CODE REVIEW

File: environments/linux-amd64-wine_osxcross [hunk 1]
```diff
@@ -74,11 +74,14 @@ RUN VERSION="6.2.1" ARCH="i386" gmp.sh
...
 
-
 RUN VERSION="3.7.2" ARCH="x86_64" gnutls.sh
 RUN VERSION="3.7.2" ARCH="i386" gnutls.sh
 
+RUN VERSION="6.2" ARCH="x86_64" ncurses.sh
+RUN VERSION="6.2" ARCH="i386" ncurses.sh
 
+RUN VERSION="2.0.12" ARCH="x86_64" SDL2.sh
+RUN VERSION="2.0.12" ARCH="i386" SDL2.sh
```
Comment (Gcenx) [inline environments/linux-amd64-wine_osxcross:85]: Bump this to the latest release then dump the now unneeded patch.

**Review by Gcenx — COMMENTED** [review]
The only remaining one now is converting over FAudio and bumping the version.

File: environments/linux-amd64-wine_osxcross (hunk 1 above)
Comment (qparis) [reply environments/linux-amd64-wine_osxcross:85]: Are you sure that the patch is now unneeded?

Comment (Gcenx) [reply environments/linux-amd64-wine_osxcross]: Yep I'm sure, the problem was resolved upstream after my report.

**Comment by Gcenx** [comment]
This seems good to me, just need to squash down the commits.

I have some commits for the build scripts later depending on the outcome of the test builds I left running.

**Comment by qparis** [comment]
Once I have tested, I squash & merge

**Pull request merged**